core/ledger/kvledger/unjoin_channel_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package kvledger
+
+import (
+	"testing"
+
+	configtxtest "github.com/hyperledger/fabric/common/configtx/test"
+	"github.com/hyperledger/fabric/core/ledger/kvledger/msgs"
+	"github.com/hyperledger/fabric/core/ledger/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnjoinChannel(t *testing.T) {
+	conf, cleanup := testConfig(t)
+	conf.HistoryDBConfig.Enabled = false
+	defer cleanup()
+
+	ledgerID := "ledger_unjoin"
+
+	provider := testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	activeLedgerIDs, err := provider.List()
+	require.NoError(t, err)
+	require.Len(t, activeLedgerIDs, 0)
+
+	genesisBlock, err := configtxtest.MakeGenesisBlock(ledgerID)
+	require.NoError(t, err)
+	_, err = provider.CreateFromGenesisBlock(genesisBlock)
+	require.NoError(t, err)
+
+	activeLedgerIDs, err = provider.List()
+	require.NoError(t, err)
+	require.Len(t, activeLedgerIDs, 1)
+	provider.Close()
+
+	// Unjoin the channel from the peer
+	err = UnjoinChannel(conf, ledgerID)
+	require.NoError(t, err)
+
+	// channel should no longer be present in the channel list
+	provider = testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	activeLedgerIDs, err = provider.List()
+	require.NoError(t, err)
+	require.Len(t, activeLedgerIDs, 0)
+
+	provider.Close()
+}
+
+// Unjoining an unjoined channel is OK.
+func TestUnjoinUnjoinedChannel(t *testing.T) {
+	conf, cleanup := testConfig(t)
+	conf.HistoryDBConfig.Enabled = false
+	defer cleanup()
+
+	ledgerID := "ledger_unjoin_unjoined"
+
+	provider := testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	genesisBlock, err := configtxtest.MakeGenesisBlock(ledgerID)
+	require.NoError(t, err)
+	_, err = provider.CreateFromGenesisBlock(genesisBlock)
+	require.NoError(t, err)
+	provider.Close()
+
+	// unjoin the channel
+	require.NoError(t, UnjoinChannel(conf, ledgerID))
+
+	provider = testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	activeLedgerIDs, err := provider.List()
+	require.NoError(t, err)
+	require.Len(t, activeLedgerIDs, 0)
+	provider.Close()
+
+	// Subsequent unjoins will not throw an error
+	require.NoError(t, UnjoinChannel(conf, ledgerID))
+	require.NoError(t, UnjoinChannel(conf, ledgerID))
+	require.NoError(t, UnjoinChannel(conf, ledgerID))
+}
+
+func TestUpdateLedgerStatus(t *testing.T) {
```

Comment:
Though it does not harm having this as a separate test but in any case, the assertions made in this tests should be made in the test `TestUnjoinChannel` above and as such this test can be removed.